sdcm/cluster_k8s/__init__.py

```diff
@@ -562,6 +567,51 @@ def deploy_scylla_operator(self, pool_name: str = None) -> None:
         # Start the Scylla Operator logging thread
         self.start_scylla_operator_journal_thread()
 
+    @log_run_info
+    def upgrade_scylla_operator(self, new_helm_repo: str,
+                                new_chart_version: str,
+                                new_docker_image: str = '') -> None:
+        LOGGER.info("Upgrade Scylla Operator using '%s' helm chart and '%s' docker image\n"
+                    "Helm repo: %s", new_chart_version, new_docker_image, new_helm_repo)
+
+        local_repo_name = "scylla-operator-upgrade"
+        LOGGER.debug(self.helm(f"repo add {local_repo_name} {new_helm_repo}"))
+        self.helm('repo update')
+
+        # Calculate new chart name if it is not specific
+        if new_chart_version in ("", "latest"):
+            new_chart_version = self.get_latest_chart_version(f"{local_repo_name}/scylla-operator")
+            LOGGER.info(
+                "Using automatically found following latest scylla-operator "
+                "upgrade chart version: %s", new_chart_version)
+        else:
+            LOGGER.info("Using following predefined scylla-operator upgrade chart version: %s",
+                        new_chart_version)
+
+        # Get existing scylla-operator helm chart values
+        values = HelmValues(json.loads(self.helm(
```

Comment:
Be aware that helm can throw warning messages that will break json structure

Reply:
Maybe. But now it works, tested. If such error appears I think it will be easy to fix knowing the specifics of additional lines.